tutorial/hipc18/cpp/futures/graph.cpp

```diff
@@ -0,0 +1,59 @@
+#include "hclib_cpp.h"
+#include <unistd.h>
+
+/*
+ * This program is computing the following task graph using async_future
+ *
+ *       (A)
+ *       /\
+ *      /  \
+ *    (B)  (C)
+ *     \    /\
+ *      \  /  \
+ *       (D)  (E)
+ *         \   /
+ *          \ /
+ *          (F)
+ */
+
+int main(int argc, char **argv) {
+  char const *deps[] = { "system" }; 
+  hclib::launch(deps, 1, [&]() {
+    hclib::future_t<void> *a = hclib::async_future([=]() {
+      sleep(1);
+      printf("A\n");
+      return;
+    });
+    hclib::future_t<void> *b = hclib::async_future([=]() {
+      a->get();
```

Comment:
This get() will not wait for Task A to finish. We should use wait() here. Similarly all the get() calls in this example.